observer.go

```diff
@@ -83,21 +112,55 @@ func NewObserver[T any](onNext func(value T), onError func(err error), onComplet
 // is provided to each callback.
 func NewObserverWithContext[T any](onNext func(ctx context.Context, value T), onError func(ctx context.Context, err error), onComplete func(ctx context.Context)) Observer[T] {
 	return &observerImpl[T]{
-		status:     0,
-		onNext:     onNext,
-		onError:    onError,
-		onComplete: onComplete,
+		status:        0,
+		capturePanics: true,
+		onNext:        onNext,
+		onError:       onError,
+		onComplete:    onComplete,
+	}
+}
+
+// NewObserverUnsafe creates a new Observer that does NOT wrap callbacks with
+// panic-recovery. Use this only in performance-sensitive paths where callers
+// guarantee no panics or want panics to propagate to the caller. This mirrors
+// the repository's "unsafe" naming for performance-optimized constructors.
+func NewObserverUnsafe[T any](onNext func(value T), onError func(err error), onComplete func()) Observer[T] {
```

Comment:
"NewObserverUnsafe" -> "NewUnsafeObserver" to match the "NewUnsafeObservable" naming.